plugins/azure/appservice/securityLogsEnabled.js

```diff
@@ -0,0 +1,88 @@
+    recommended_action: 'Modify Web Apps and enable diagnostic settings for all logs.',
+    link: 'https://learn.microsoft.com/en-us/azure/app-service/overview-monitoring',
+    apis: ['webApps:list', 'diagnosticSettings:listByAppServices'],
+    settings: {
+        diagnostic_logs: {
+            name: 'Diagnostic Logs Enabled',
+            description: 'Comma separated list of diagnostic logs that should be enabled at minimum i.e. AppServiceAntivirusScanAuditLogs, AppServiceHTTPLogs etc. If you have enabled allLogs, then resource produces pass result. If you only want to check if logging is enabled or not, irrespecitve of log type, then add * in setting.',
+            regex: '^.*$',
+            default: 'AppServiceAntivirusScanAuditLogs, AppServiceHTTPLogs, AppServiceConsoleLogs, AppServiceAppLogs, AppServiceFileAuditLogs,AppServiceAuditLogs, AppServiceIPSecAuditLogs, AppServicePlatformLogs'
+        },
+    },
+    realtime_triggers: ['microsoftweb:sites:write', 'microsoftinsights:diagnosticsettings:write', 'microsoftinsights:diagnosticsettings:delete', 'microsoftweb:sites:delete'],
+
+    run: function(cache, settings, callback) {
+        const results = [];
+        const source = {};
+        const locations = helpers.locations(settings.govcloud);
+
+        var config = {
+            diagnostic_logs: settings.diagnostic_logs || this.settings.diagnostic_logs.default,
+        };
+
+        async.each(locations.webApps, (location, rcb) => {
+
+            const webApps = helpers.addSource(cache, source,
+                ['webApps', 'list', location]);
+
+            if (!webApps) return rcb();
+
+            if (webApps.err || !webApps.data) {
+                helpers.addResult(results, 3, 'Unable to query for Web Apps: ' + helpers.addError(webApps), location);
+                return rcb();
+            }
+
+            if (!webApps.data.length) {
+                helpers.addResult(results, 0, 'No existing Web Apps found', location);
+                return rcb();
+            }
+
+            webApps.data.forEach(webApp => {
+                if (!webApp.id) return;
+                if (webApp && webApp.kind && webApp.kind.startsWith('functionapp')) return;
+                const diagnosticSettings = helpers.addSource(cache, source,
+                    ['diagnosticSettings', 'listByAppServices', location, webApp.id]);
+                if (!diagnosticSettings || diagnosticSettings.err || !diagnosticSettings.data) {
+                    helpers.addResult(results, 3, `Unable to query for App Service diagnostic settings: ${helpers.addError(diagnosticSettings)}`,
+                        location, webApp.id);
+                    return;
+                }
+                var found = true;
+                var missingLogs = [];
+                if (config.diagnostic_logs == '*') {
```

Comment:
```suggestion
                if (!webApp.id) return;
                
                if (webApp && webApp.kind && webApp.kind.startsWith('functionapp')) return;
                
                const diagnosticSettings = helpers.addSource(cache, source,
                    ['diagnosticSettings', 'listByAppServices', location, webApp.id]);
                    
                if (!diagnosticSettings || diagnosticSettings.err || !diagnosticSettings.data) {
                    helpers.addResult(results, 3, `Unable to query for App Service diagnostic settings: ${helpers.addError(diagnosticSettings)}`,
                        location, webApp.id);
                    return;
                }
                
                var found = true;
                var missingLogs = [];
                
                if (config.diagnostic_logs == '*') {
```
add line break before/after each if